### ISSUE TITLE
Consider using "OpenTracing.Contrib.NetCore" in examples

### ISSUE THREAD

**Issue by cwe1ss** [jaegertracing/jaeger-client-csharp#14]
What do you think about using [OpenTracing.Contrib.NetCore](https://github.com/opentracing-contrib/csharp-netcore) in the examples project? It does not require people to add a custom middleware and it has more features. This would give people better guidance on what to use in ASP.NET Core projects.

Your examples could then focus on how the LetsTrace tracer is initialized/configured. As it implements OpenTracing, usage in instrumentation-code isn't tracer-specific anyway.

**Comment by grounded042** [comment]
Good idea. I'll take a look.

**Comment by Falco20019** [comment]
One main problem we have, is that flushing for the `RemoteReporter` is right now only done on disposal. But since `ITracer` is not implementing `IDisposable` it's not called when the application is stopping. In my [java oriented implementation](https://github.com/Falco20019/jaeger-client-csharp/blob/dev/JaegerClientCsharp/Jaeger/Reporters/RemoteReporter.cs), I had a peridoc flush task. We need to add this here too. I wanted to improve the `RemoteReporter` anyway. So for now it's better to use a transient `IJaegerCoreTracer` instance, since those will be disposed cleanly.

As a workaround, I'm using the following line in `Configure`, but this also only works if the application is shutdown clean and not by closing the browser:
```csharp
public void Configure(IApplicationBuilder app, IHostingEnvironment env, IApplicationLifetime applicationLifetime)
{
    // ...

    // Register tracer for disposal:
    // TODO: This is NOT triggered when the browser is stopped. Only when closing by CTRL-C...
    applicationLifetime.ApplicationStopping.Register(() => Tracer.Dispose());
}
```

But I will also look into using `OpenTracing.Contrib.NetCore` in the `opentracing-tutorial`.

**Comment by Falco20019** [comment]
This has been resolved by #55.
